### DISCUSSION TITLE
[2023-10-09] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by MichaelMauderer** [enso-org/enso#7996] [Weekly Demo Day]
* Finished work on expression updates. Also added a node output type indicator to show some information received.   #7982
 
https://github.com/enso-org/enso/assets/1428930/31ffff78-ff2c-4e0b-bcde-ddc507cc0226

* Worked on hooking up the Play Button and Execution Mode Selector with the language server API. Almost done, just pending clean-up and testing.  #7990

**Comment by somebody1234** [comment]
PRs merged:
- #7873
  - Includes various fixes by @Frizi 

Main focus:
- #7945

https://github.com/enso-org/enso/assets/4046547/97fc2ceb-f732-456e-8e27-4c58860e33fa

Miscellaneous small PRs:
- #7986
- #7988
- #8004

**Comment by farmaazon** [comment]
Worked on #7784 

Worked with AST to make proper input interpretation. Added several helper functions. Not working yet, but I'm close.

**Comment by Frizi** [comment]
- fixed some edge cases in text edit translation logic https://github.com/enso-org/enso/issues/7967
- improved reliability of engine connection and state transitions, added retries on timeouts and improved logging https://github.com/enso-org/enso/issues/7995
- Added dimming to nodes during evaluation and loaders to visualizations
<img width="209" alt="image" src="https://github.com/enso-org/enso/assets/919491/1836c892-1e5e-4eff-adbb-49ffd9ab5e92">

- Improved visuals of code editor, added resizing

https://github.com/enso-org/enso/assets/919491/da052dd1-a60a-46ca-926f-de4ff4fa2091

- Currently working on using real Enso AST within the graph view. https://github.com/enso-org/enso/issues/7791

**Comment by vitvakatu** [comment]
I finally finished all tasks presented in recent weeks:
- https://github.com/enso-org/enso/pull/7844
- https://github.com/enso-org/enso/pull/7902
- https://github.com/enso-org/enso/pull/7779

Currently working on documentation panel for the new IDE (https://github.com/enso-org/enso/issues/7924),
WIP screenshot with some mocked data:

<img width="622" alt="Screenshot 2023-10-09 at 6 14 24 PM" src="https://github.com/enso-org/enso/assets/6566674/67109fb3-2e64-4acf-a28e-193cb2b6f080">

**Comment by farmaazon** [comment]
And here is the big demo of the entire GUI promised last week:


https://github.com/enso-org/enso/assets/3919101/fb2416ae-f729-4b2a-8a84-41d9b5316930



Important notice: to have it so flawlessly working, you must remove old metadata (even from freshly created project) leaving at least 3 newlines at the end.

**Comment by mwu-tow** [comment]
* The build script support for new GUI/IDE landed (https://github.com/enso-org/enso/pull/7832) — CI checks the new GUI project and the new IDE is uploaded as a CI run artifact: 
![obraz](https://github.com/enso-org/enso/assets/1548407/96b71712-604c-4f67-993e-964fcb38081e)
* Spent quite much time on CI issues.
  * The issue with AppImage should be now resolved. (Thanks to Dmitrii!).
* Working on a follow-up build script update that will upload new IDE builds to the release assets.

**Comment by wdanilo** [comment]
We have finally a huge progress with website. What has been done:

- The top navigation floating panel is now functioning, allowing to scroll the website by clicking.
- Website has anchors, so you can send a link to people to a particular section, like `enso.org/#virtual-data-depot`.
- There is a buy button now which redirects to empty website, but we are working on putting pricing there right now.
- We have really nice nodes and visualization animations. They are crucial to show the users how the product works. For example, if user selects "parse", they should see how things are being parsed. If you are used to the concept, it's obvious, but when you see the product/animation for the first time, you need a guidance to understand it and time to digest it.
- The animation framework to create nodes is really nice now - here is a part of the animation definition:

```js

        transformNode(
          () => WORKBENCH.node1.current,
          stage1_node1,
          stage2_node1
        ),

        createNode(() => WORKBENCH.node2.current, {
          vizData: fileData.shopLocations(),
          afterTypingEnd: [
            {
              fn: (t) => {
                WORKBENCH.node2.current.modifyVizData((data) => {
                  const opacity = 1 - t * 0.5;
                  data.meta.columnOpacity = [
                    opacity,
                    opacity,
                    opacity,
                    opacity,
                  ];
                });
              },
            },
          ],
        }),

        selectOptions(
          () => WORKBENCH.node2.current,
          [1],
          [
            {
              add: '"Latitude"',
              fn: (t) => {
                WORKBENCH.node2.current.modifyVizData((data) => {
                  const opacity = 0.5 + t * 0.5;
                  data.meta.columnOpacity[2] = opacity;
                });
              },
            },
            {
              add: '"Longitude"',
              fn: (t) => {
                WORKBENCH.node2.current.modifyVizData((data) => {
                  const opacity = 0.5 + t * 0.5;
                  data.meta.columnOpacity[3] = opacity;
                });
              },
            },
          ]
        ),
```


Things to be done below. Based on the velocity last week, I'm very positive to have it this week:
- Rest of node animations (4 missing sections).
- Pricing page.
- Linking to example videos.
- Connecting chat.
- Fixing bugs on mobile.



https://github.com/enso-org/enso/assets/1623053/6043f4b1-f33a-4060-a528-ad7f421e0857

**Comment by JaroslavTulach** [comment]
Engine team is working on stability & reliability (including supporting new IDE requirements):
- #7898
- #7985
- #7868
- #7953
- #7943
- #7994
- #7970
- #7937
- Loaded `y-websocket` server: https://github.com/enso-org/enso/issues/7954#issuecomment-1746722284

**Comment by kazcw** [comment]
I've been finishing #7884--the last part remaining is finishing resolution of a longstanding TODO to produce complete span info in the parser (#5449). At this point there are a few known unhandled cases, but most nodes should be working and I don't think any other changes will be needed on the TS side, so the PR should be ready to start integration; I will complete #5449 and write more comprehensive tests in a follow-up PR.
